### PR TITLE
doc: add issue and PR templates

### PR DESCRIPTION
**Pull request by andyzhangx**
It's better have a template for issues and PRs, we could refine the templates later.
/assign @jdef @saad-ali

### CODE REVIEW

**Comment by jdef** [comment]
Thanks for the PR.

The templates seem a bit k8s-centric, and this is not a k8s project. Nor does it use k8s PR robots.

I'm open to the idea of templates, but I think we should probably scrub e.g. links to k8s-specific things.

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+**What you expected to happen**:
+
+**How to reproduce it**:
+
+**Anything else we need to know?**:
+
+**Environment**:
+- CSI Spec version:
+- Kubernetes version (use `kubectl version`):
```
Comment (xing-yang) [inline .github/ISSUE_TEMPLATE/bug-report.md:21]: Can you remove this?

File: .github/ISSUE_TEMPLATE/support.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+about: Ask questions about this project
+
+---
+
+<!-- 
+STOP -- PLEASE READ!
+
+GitHub is not the right place for support requests.
+
+If you're looking for help, post your question on the [Kubernetes Slack ](http://slack.k8s.io/) Sig-Storage Channel.
```
Comment (xing-yang) [inline .github/ISSUE_TEMPLATE/support.md:12]: CSI spec is not a project in K8s.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<!--  Thanks for sending a pull request!  Here are some tips for you:
+
+1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
+2. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
+3. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
+-->
```
Comment (xing-yang) [inline .github/PULL_REQUEST_TEMPLATE.md:6]: These comments are very K8s specific.

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 1 above)
Comment (andyzhangx) [reply .github/ISSUE_TEMPLATE/bug-report.md]: fixed

File: .github/ISSUE_TEMPLATE/support.md (hunk 2 above)
Comment (andyzhangx) [reply .github/ISSUE_TEMPLATE/support.md]: removed

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (andyzhangx) [reply .github/PULL_REQUEST_TEMPLATE.md]: removed

**Comment by andyzhangx** [comment]
I have removed the k8s related things, PTAL, thanks.

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+<!--
+*Automatically closes linked issue when PR is merged.
+Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
+_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
+-->
+Fixes #
+
+**Special notes for your reviewer**:
+
+**Does this PR introduce a user-facing change?**:
```
Comment (jdef) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: maybe s/user-facing/API-breaking/ ?

CSI doesn't have "users" exactly, but API consumers are keenly interested in possible API breakage

Comment (andyzhangx) [reply .github/PULL_REQUEST_TEMPLATE.md]: fixed.

**Review by jdef — APPROVED** [review]

**Pull request merged**